Original patch from gentoo: gentoo/src/patchsets/gcc/4.3.0/uclibc/90_all_306-libstdc++-namespace.patch
diff -rup gcc-4.2.orig/libstdc++-v3/config/locale/uclibc/messages_members.h gcc-4.2/libstdc++-v3/config/locale/uclibc/messages_members.h
diff -durN gcc-4.3.0.orig/libstdc++-v3/config/locale/uclibc/messages_members.h gcc-4.3.0/libstdc++-v3/config/locale/uclibc/messages_members.h
--- gcc-4.3.0.orig/libstdc++-v3/config/locale/uclibc/messages_members.h	2008-06-10 14:45:50.000000000 +0200
+++ gcc-4.3.0/libstdc++-v3/config/locale/uclibc/messages_members.h	2008-06-10 14:46:05.000000000 +0200
@@ -32,7 +32,8 @@
 //
 
 // Written by Benjamin Kosnik <bkoz@redhat.com>
-
+namespace std
+{
 #ifdef __UCLIBC_MJN3_ONLY__
 #warning fix prototypes for *textdomain funcs
 #endif
@@ -115,3 +116,4 @@
 	   this->_S_create_c_locale(this->_M_c_locale_messages, __s); 
 	 }
      }
+}
diff -durN gcc-4.3.0.orig/libstdc++-v3/config/locale/uclibc/time_members.h gcc-4.3.0/libstdc++-v3/config/locale/uclibc/time_members.h
--- gcc-4.3.0.orig/libstdc++-v3/config/locale/uclibc/time_members.h	2008-06-10 14:45:50.000000000 +0200
+++ gcc-4.3.0/libstdc++-v3/config/locale/uclibc/time_members.h	2008-06-10 14:46:05.000000000 +0200
@@ -33,7 +33,8 @@
 //
 
 // Written by Benjamin Kosnik <bkoz@redhat.com>
-
+namespace std
+{
   template<typename _CharT>
     __timepunct<_CharT>::__timepunct(size_t __refs) 
     : facet(__refs), _M_data(NULL), _M_c_locale_timepunct(NULL), 
@@ -74,3 +75,4 @@
       delete _M_data; 
       _S_destroy_c_locale(_M_c_locale_timepunct); 
     }
+}
